### ISSUE TITLE
Should CORS be abandoned?

### ISSUE THREAD

**Issue by jackzhp** [monsur/enable-cors.org#130]
I think that CORS design is completely unnecessary, stupid, and so should be abandoned. I have laid down my arguments, if you think you know this issue pretty well, would you please look at it, and comment it.

https://github.com/jackzhp/CORS-should-retire
or
http://blog.sina.com.cn/s/blog_93b70ae70102wxe8.html

Hope you can contribute to the change!

**Comment by divbzero** [comment]
@jackzhp's suggestion doesn't address a core purpose of the same-origin policy: to prevent malicious websites from reading sensitive data on other websites.

**Comment by dubahu** [comment]
I think you need to understand the purpose of CORS and what it does (and prevents) before making that statement.  If you set an open CORS policy, I can reproduce your website on a different URL using your source CSS and JS, and also using your bandwidth to server those files.  For you, that might not be problematic, but it sure is for government and banking sites for example.  It's been proven time and time again that users don't pay enough attention to URLs and with an open CORS policy, a bad actor can even use your actual assets on a malicious site that is designed to spread fake government information or steal banking data.  CORS is very easy to work with, it just requires some basic learning.

**Comment by sideshowbarker** [comment]
> If you set an open CORS policy, I can reproduce your website on a different URL using your source CSS and JS, and also using your bandwidth to server those files. 

You can do that even if the site doesn’t have an open CORS policy.

**Comment by dubahu** [comment]
> > If you set an open CORS policy, I can reproduce your website on a different URL using your source CSS and JS, and also using your bandwidth to server those files.
> 
> You can do that even if the site doesn’t have an open CORS policy.

No, no you cannot.  It's literally the thing that CORS prevents.  The cross origin request will be blocked in modern browsers unless a CORS policy is set which allows it.  That would be allowing the origin explicitly, or allowing all origins, as recommended here.  Which is why you shouldn't recommend it blindly as a solution.

**Comment by monsur** [comment]
Closing old comments, plus this isn't the purpose of the site. Take it to the W3C.

**Comment by dubahu** [comment]
Lol this was 6 years ago.   Why now?

________________________________

**Comment by monsur** [comment]
Haha, AI! https://bsky.app/profile/monsur.hossa.in/post/3mcgrhcfcy22u
